### PR TITLE
Do not gzip assets that are already gzipped

### PR DESCRIPTION
**Pull request by byroot**
I know it's not really the vanilla use case for sprockets, but we were using it to serve a few `tar.gz` archives.

Turns out sprockets would gzip them a second time :/

@josh do you think that fix make sense? I can also take a look at the 3.x branch to see if it's also affected.

cc @ryanbalsdon for discovering the bug.

### CODE REVIEW

**Comment by rafaelfranca** [comment]
@byroot sprockets 3 don't gzip the assets anymore, so it will only affect 2.x.

**Comment by byroot** [comment]
Awesome!

File: lib/sprockets/asset.rb [hunk 1]
```diff
@@ -139,7 +139,9 @@ def stale?(environment)
     # Save asset to disk.
     def write_to(filename, options = {})
       # Gzip contents if filename has '.gz'
-      options[:compress] ||= File.extname(filename) == '.gz'
```
Comment (byroot) [inline lib/sprockets/asset.rb]: By the way I think there was a bug here. You couldn't pass `compress: false`

**Comment by rafaelfranca** [comment]
I had this same bug before, so I think it makes sense.

**Pull request merged**